app/src/main/java/org/oppia/android/app/splash/SplashActivityPresenter.kt

```diff
@@ -265,36 +279,142 @@ class SplashActivityPresenter @Inject constructor(
-        activity.startActivity(OnboardingActivity.createOnboardingActivity(activity))
+        launchOnboardingActivity()
+      }
+    }
+  }
+
+  private fun handleUserOnboarded() {
+    if (enableOnboardingFlowV2.value) {
+      getProfileOnboardingState()
+    } else {
+      activity.startActivity(ProfileChooserActivity.createProfileChooserActivity(activity))
+      activity.finish()
+    }
+  }
+
+  private fun getProfileOnboardingState() {
+    profileManagementController.getProfileOnboardingMode().toLiveData().observe(
+      activity,
+      { result ->
+        when (result) {
+          is AsyncResult.Success -> computeLoginRoute(result.value)
+          is AsyncResult.Failure -> oppiaLogger.e(
+            "SplashActivity",
+            "Encountered unexpected non-successful result when fetching onboarding state",
+            result.error
+          )
+          is AsyncResult.Pending -> {}
+        }
+      }
+    )
+  }
+
+  private fun computeLoginRoute(onboardingMode: ProfileOnboardingMode) {
+    when (onboardingMode) {
+      ProfileOnboardingMode.NEW_INSTALL -> {
+        launchOnboardingActivity()
+      }
+      ProfileOnboardingMode.SOLE_LEARNER_PROFILE_ONLY -> fetchProfile()
+      else -> {
+        activity.startActivity(ProfileChooserActivity.createProfileChooserActivity(activity))
         activity.finish()
       }
     }
   }
 
+  private fun fetchProfile() {
+    profileManagementController.getProfiles().toLiveData().observe(activity) { result ->
+      when (result) {
+        is AsyncResult.Success -> handleProfiles(result.value)
+        is AsyncResult.Failure -> oppiaLogger.e(
+          "SplashActivity", "Failed to retrieve the list of profiles", result.error
+        )
+        is AsyncResult.Pending -> {} // no-op
+      }
+    }
+  }
+
+  private fun handleProfiles(profiles: List<Profile>) {
+    val soleLearnerProfile = profiles.find { it.profileType == ProfileType.SOLE_LEARNER }
+    if (soleLearnerProfile != null) {
+      proceedBasedOnProfileState(soleLearnerProfile)
+    } else {
+      launchOnboardingActivity()
+    }
+  }
+
+  private fun proceedBasedOnProfileState(profile: Profile) {
+    when {
+      profile.startedProfileOnboarding && !profile.completedProfileOnboarding -> {
+        resumeOnboarding(profile.id, profile.name)
+      }
+      profile.startedProfileOnboarding && profile.completedProfileOnboarding -> {
+        loginToProfile(profile.id)
+      }
+      else -> launchOnboardingActivity()
+    }
+  }
+
+  private fun resumeOnboarding(profileId: ProfileId, profileName: String) {
+    val introActivityParams = IntroActivityParams.newBuilder()
+      .setProfileNickname(profileName)
+      .build()
+
+    val intent = IntroActivity.createIntroActivity(activity).apply {
+      putProtoExtra(PARAMS_KEY, introActivityParams)
+      decorateWithUserProfileId(profileId)
+    }
+
+    activity.startActivity(intent)
+  }
+
+  private fun loginToProfile(profileId: ProfileId) {
```

Comment:
```suggestion
  private fun logInToProfile(profileId: ProfileId) {
```

Nit for consistency & correct verb tense. Ditto elsewhere in the PR.

Reply:
Updated -- other occurances previously updated in profileChooser and AudioLanguageFragment.